src/components/views/rooms/DecryptionFailureBar.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { Room } from 'matrix-js-sdk/src/models/room';
+import { MatrixEvent } from 'matrix-js-sdk/src/models/event';
+
+import Modal from '../../../Modal';
+import { _t } from '../../../languageHandler';
+import defaultDispatcher from "../../../dispatcher/dispatcher";
+import { Action } from "../../../dispatcher/actions";
+import AccessibleButton from '../elements/AccessibleButton';
+import { OpenToTabPayload } from "../../../dispatcher/payloads/OpenToTabPayload";
+import { UserTab } from "../dialogs/UserTab";
+import MatrixClientContext from "../../../contexts/MatrixClientContext";
+import SetupEncryptionDialog from '../dialogs/security/SetupEncryptionDialog';
+import { SetupEncryptionStore } from '../../../stores/SetupEncryptionStore';
+import Spinner from "../elements/Spinner";
+
+interface IProps {
+    room: Room;
+    failures: MatrixEvent[];
+}
+
+// Number of milliseconds to display a loading spinner before prompting the user for action
+const WAIT_PERIOD = 5000;
+
+export const DecryptionFailureBar: React.FC<IProps> = ({ failures, room }) => {
+    const context = useContext(MatrixClientContext);
+
+    const [waiting, setWaiting] = useState<boolean>(true);
```

Comment:
more comments please. Could you document what each of the state properties mean?